### ISSUE TITLE
Elixir 1.6.1 produces compilation warnings

### ISSUE THREAD

**Issue by dragosdm** [rrrene/credo#500]
### Environment

* Credo version (`mix credo -v`): 0.9.0-rc2
* Erlang/Elixir version (`elixir -v`): Elixir 1.6.1
* Operating system: OSX 10.13.3

### What were you trying to do?
`MIX_ENV=test mix do compile --warnings-as-errors, test --trace, credo --strict`

### Expected outcome
Compilation succeeds without any warnings.

### Actual outcome

```
Checking 202 source files (this might take a while) ...
warning: found ? followed by codepoint 0x20 (space), please use \s instead
  nofile:1

warning: found ? followed by codepoint 0x20 (space), please use \s instead
  nofile:1

warning: found ? followed by codepoint 0x20 (space), please use \s instead
  nofile:1
```

EDIT:
I did a sweep through the entire project structure and finally found the file that was making credo throw the above error

```
defmacro some_macro(type) do
    quote do
      fragment("CASE
        WHEN
          (? = 'some_text'
          OR ? = 'some_text_2'
          OR ? = 'some_text_3')
        THEN '1'
        ELSE '2'
      END", unquote(type), unquote(type), unquote(type))
    end
  end
```

**Comment by rrrene** [comment]
@dragosdm Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by vanjabucic** [comment]
Still seeing warnings with version 1.0.0
```
warning: found ? followed by codepoint 0x20 (space), please use ?\s instead
  nofile:134
```

file content I believe this is coming from:
```
@update_tag_stmt """
    UPDATE TBL SET [some fields], last_updated=?
    WHERE key1=? AND key2=? and key3=?
    IF EXISTS
  """
```
Linux, Erlang 20, Elixir 1.7.4

**Comment by vanjabucic** [comment]
Still seeing the warning with the heredoc mentioned above.
Should the issue be reopened?

**Comment by rrrene** [comment]
@vanjabucic Sure. 

I am not sure what causes this. When I create a project with a single module that jut contains the module attribute you mentioned, I do not get this warning with Credo v1.0.3. 

Could you try to post a minimal example to reproduce this?

**Comment by scouten** [comment]
FWIW I am still seeing something like this. Given that the warning is reported in `nofile`, I don't really know where to start looking for a minimal repro case. Now that my project is available in open source, I can show you the specific place where I'm seeing it. The error reproduces here:

https://travis-ci.org/elixir-git/xgit/builds/537335447

**Comment by rrrene** [comment]
@scouten That message comes from [Elixir's tokenizer](https://github.com/elixir-lang/elixir/blob/f44adfbeb01ba315be53ebbbe4e30007d83ffdce/lib/elixir/src/elixir_tokenizer.erl#L239) while reading `xgit`'s source.

I will have a look if we can do a better of pointing out the file which causes the issue :+1:

**Comment by rrrene** [comment]
Hi, thx again for this. I am closing it now, because Credo is now [giving you the name of the file](https://github.com/rrrene/credo/blob/f5d7bb76c967e23c5fda8c82f427de6fe6825dc7/lib/credo/code.ex#L137).

If you disagree, please re-open and elaborate! :+1:
